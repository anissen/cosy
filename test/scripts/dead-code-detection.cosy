fn f() {
    print 'before return'
    if (false) return _
    print 'after first return'
    return _
    print 'after second return'
}

f()

for true {
    print 'before break'
    break
    print 'after break'
}

for 0..3 {
    print 'before continue'
    continue
    print 'after continue'
}

if 5 < 4 {
    print 'always false'
} else if 4 < 5 {
    print 'always true'
} else {
    print 'never hit'
}