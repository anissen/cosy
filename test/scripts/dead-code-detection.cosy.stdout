■ test/scripts/dead-code-detection.cosy, line 5:
3 |     if (false) return _
4 |     print 'after first return'
5 |     return _
        ^^^^^^ Unreachable code after return statement.
6 |     print 'after second return'
7 | }

■ test/scripts/dead-code-detection.cosy, line 13:
11 | for true {
12 |     print 'before break'
13 |     break
         ^^^^^ Unreachable code after return statement.
14 |     print 'after break'
15 | }

■ test/scripts/dead-code-detection.cosy, line 19:
17 | for 0..3 {
18 |     print 'before continue'
19 |     continue
         ^^^^^^^^ Unreachable code after return statement.
20 |     print 'after continue'
21 | }

■ test/scripts/dead-code-detection.cosy, line 23:
21 | }
22 | 
23 | if 5 < 4 {
     ^^ This condition is always false.
24 |     print 'always false'
25 | } else if 4 < 5 {

■ test/scripts/dead-code-detection.cosy, line 25:
23 | if 5 < 4 {
24 |     print 'always false'
25 | } else if 4 < 5 {
            ^^ This condition is always true.
26 |     print 'always true'
27 | } else {

